providers/amazon/tests/unit/amazon/aws/operators/test_sns.py

```diff
@@ -65,20 +69,29 @@ def test_init(self):
         assert op.hook._config.read_timeout == 42
 
     @mock.patch.object(SnsPublishOperator, "hook")
-    def test_execute(self, mocked_hook):
+    @pytest.mark.parametrize(
+        "message_deduplication_id_,message_group_id_",
+        [
+            ("abc", "a"),
+            (None, None),
```

Comment:
Could add the two cases where one is none and the other has a value, and vice versa.